src/leo_s3_user.erl

```diff
@@ -192,39 +195,30 @@ import(UserId, AccessKey, SecretKey) ->
 
 
 %% @private
-import_1(UserId, AccessKey, SecretKey) ->
+import_1(UserId, AccessKey, SecretKey, Password, RoleId) ->
     CreatedAt = leo_date:now(),
 
     %% insert a user-credential-info
-    case mnesia:write(?AUTH_TABLE,
+    ok = mnesia:write(?AUTH_TABLE,
                       #credential{access_key_id = AccessKey,
                                   secret_access_key = SecretKey,
-                                  created_at = CreatedAt}, write) of
-        ok ->
-            %% insert a user-info
-            case mnesia:write(?USERS_TABLE,
-                              #?S3_USER{id = UserId,
-                                        password = <<>>,
-                                        created_at = CreatedAt,
-                                        updated_at = CreatedAt}, write) of
-                ok ->
-                    %% insert a user-credential-info
-                    case mnesia:write(?USER_CREDENTIAL_TABLE,
-                                      #user_credential{user_id = UserId,
-                                                       access_key_id = AccessKey,
-                                                       created_at = CreatedAt}, write) of
-                        ok ->
-                            {ok, [{access_key_id,     AccessKey},
-                                  {secret_access_key, SecretKey}]};
-                        _ ->
-                            {error, transaction_abort}
-                    end;
-                _ ->
-                    {error, transaction_abort}
-            end;
-        _ ->
-            {error, transaction_abort}
-    end.
+                                  created_at = CreatedAt}, write),
+
+    %% insert a user-info
+    ok = mnesia:write(?USERS_TABLE,
+                      #?S3_USER{id = UserId,
+                                password = Password,
+                                role_id = RoleId,
+                                created_at = CreatedAt,
+                                updated_at = CreatedAt}, write),
+
+    %% insert a user-credential-info
+    ok = mnesia:write(?USER_CREDENTIAL_TABLE,
+                      #user_credential{user_id = UserId,
+                                       access_key_id = AccessKey,
+                                       created_at = CreatedAt}, write),
```

Comment:
Ditto.
I think mnesia:write called inside https://github.com/leo-project/leofs/blob/v1/apps/leo_manager/src/leo_manager_console.erl#L2163-L2165 leo_s3_user:update/1 also should be included in this transaction to make the whole mnesia operation atomic. (The password field should be updated to SecretAccessKey)